pilot/pkg/networking/core/v1alpha3/accesslog_test.go

```diff
@@ -104,7 +103,7 @@ func TestListenerAccessLog(t *testing.T) {
 		tc := tc
 		t.Run(tc.name, func(t *testing.T) {
 			// Update MeshConfig
-			m := mesh.DefaultMeshConfig()
+			m := &meshconfig.MeshConfig{}
```

Comment:
I am not sure I understand this. Can you explain why this fixes it?

It seems like the issue is that we have a shared cache, so maybe we just need `accessLogBuilder.reset()` before the test (or maybe in `buildListeners`, etc)?

Reply:
IIUC, this is not to mutate the default config. I do not look into the original issue, but this way it makes no harm.

@howardjohn to check whether it fixes original issue?

Reply:
yes. The issue is we are mutating default config in the test which other tests are dependent on.

Reply:
I have run multiple times with shuffle on - it did not fail.

Reply:
How are we mutating it? DefaultMeshConfig is a new copy.

I know this fixes it but want tounderstand why

Reply:
> DefaultMeshConfig is a new copy.

Ah .. Let me debug a bit a more on how this is working :-)

Reply:
Ok. this is actually not working. When we pass `&meshconfig.MeshConfig{}`, no listeners are constructed as `proxyListenPort` is zero. So it appears to pass but is incorrect. I fixed it now. Basically the accesslog cache  should be reset before buildListener call not after.